types/index.d.ts

```diff
@@ -0,0 +1,31 @@
+// Minimum TypeScript Version: 3.0
+import {
+  Buffer,
+  BufferEncoding,
+  SyntaxExtension,
+  Token
+} from 'micromark/dist/shared-types'
+import {Root} from 'mdast'
+import {Type} from 'micromark/dist/constant/types'
+
+export = fromMarkdown
+
+declare namespace fromMarkdown {}
+
+interface MdastExtension {
+  enter: Record<Type, (token: Token) => void>
+  exit: Record<Type, (token: Token) => void>
+}
+
+interface Options {
+  extensions?: SyntaxExtension[]
+  mdastExtensions?: MdastExtension[]
+}
+
+declare function fromMarkdown(value: string | Buffer, options?: Options): Root
+
+declare function fromMarkdown(
+  value: string | Buffer,
+  encoding?: BufferEncoding,
+  options?: Options
+): Root
```

Comment:
apologies I may have been unclear before, extra interfaces that we want exported when using `export =` need to be inside the namespace that matches the export.

```suggestion
declare namespace fromMarkdown {
  interface MdastExtension {
    enter: Record<Type, (token: Token) => void>
    exit: Record<Type, (token: Token) => void>
  }

  interface Options {
    extensions?: SyntaxExtension[]
    mdastExtensions?: MdastExtension[]
  }
}

declare function fromMarkdown(value: string | Buffer, options?: fromMarkdown.Options): Root

declare function fromMarkdown(
  value: string | Buffer,
  encoding?: BufferEncoding,
  options?: fromMarkdown.Options
): Root
```